frontend/pages/hosts/details/DeviceUserPage/DeviceUserPage.tsx

```diff
@@ -232,6 +236,11 @@ const DeviceUserPage = ({
     },
     [showPolicyDetailsModal, setShowPolicyDetailsModal, setSelectedPolicy]
   );
+
+  const toggleMacSettingsModal = useCallback(() => {
```

Comment:
Do you need to use a callback here? Maybe try this instead:

```
`setShowMacSettingsModal((prevState) => !prevState)`
```

Reply:
this was just a pattern I saw elsewhere and followed in lieu of any other input. I'll do that.

Reply:
@gillespi314 I'm actually now remembering, @ghernandez345 doesn't like directly passing `set{State}` s to other components and prefers a handler of some sort, so this is in line with that. I am curious as to why he prefers that though?

Reply:
Yeah, I've seen both approaches and defer to Gabe if his preference is to always define handlers. 

In that case you can still avoid the `useCallback`, I think. 

```
const toggleMacSettingsModal = () => setShowMacSettingsModal((prevState) => !prevState)
```

Reply:
`useState` guarantees that the setter is referentially [stable](https://reactjs.org/docs/hooks-reference.html#usestate).

> React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.

But the `toggleMacSettingsModal` handler function will get recreated each render. And I see that it get passed as a prop to child components. This type of set up can result in performance problems where the child component needs intensive calculations to render. https://beta.reactjs.org/reference/react/useCallback#skipping-re-rendering-of-components. So you can add `useCallback` here to cache this particular handler prop although it might be [overkill](https://beta.reactjs.org/reference/react/useCallback#should-you-add-usecallback-everywhere). Also note that other prop changes can still trigger the child to re-render so this alone isn't a silver bullet for performance problems if the other props aren't being tightly controlled as well.